actions/drain_node_handler.go

```diff
@@ -39,10 +44,29 @@ func newDrainNodeHandler(log logrus.FieldLogger, clientset kubernetes.Interface)
 			podDeleteRetryDelay:           5 * time.Second,
 			podEvictRetryDelay:            5 * time.Second,
 			podsTerminationWaitRetryDelay: 10 * time.Second,
+			castNamespace:                 castNamespace,
+			skipDeletedTimeoutSeconds:     60,
 		},
 	}
 }
 
+// getDrainTimeout returns drain timeout adjusted to action creation time.
+// After 2 minutes grace period action timeout is adjusted using the time passed between action creation and now
+// the result is clamped between 60s and the requested timeout.
+func (h *drainNodeHandler) getDrainTimeout(action *castai.ClusterAction) time.Duration {
+	timeSinceCreated := time.Since(action.CreatedAt)
+	drainTimeout := time.Duration(action.ActionDrainNode.DrainTimeoutSeconds) * time.Second
+	// Allow to have 2 minutes grace period for newer actions.
+	if timeSinceCreated < 2*time.Minute {
```

Comment:
I believe we shouldn't have this `if` statement. This was the reason that we were cycling the same 2 minutes timeout all the time when controller was being restarted every 30 seconds.
Actual timeout should be a function:
`actualTimeout := drainTimeout - timeSinceCreated`
This could be rounded to 10 seconds intervals to account for roundtrips.

The only caveat here is if actions priority doesn't pick up this action in time due to action priorities.